WordPress/src/main/java/org/wordpress/android/ui/comments/CommentDetailFragmentAdapter.java

```diff
@@ -0,0 +1,97 @@
+package org.wordpress.android.ui.comments;
+
+import android.app.Fragment;
+import android.app.FragmentManager;
+import android.os.Bundle;
+import android.os.Parcelable;
+import android.support.v13.app.FragmentStatePagerAdapter;
+
+import org.wordpress.android.fluxc.model.CommentModel;
+import org.wordpress.android.fluxc.model.SiteModel;
+import org.wordpress.android.models.CommentList;
+import org.wordpress.android.util.AppLog;
+
+public class CommentDetailFragmentAdapter extends FragmentStatePagerAdapter {
+
+    private final SiteModel mSite;
+    private final CommentAdapter.OnLoadMoreListener mOnLoadMoreListener;
+    private final CommentList mCommentList;
+
+    CommentDetailFragmentAdapter(FragmentManager fm,
+                                 CommentList commentList,
+                                 SiteModel site,
+                                 CommentAdapter.OnLoadMoreListener onLoadMoreListener) {
+        super(fm);
+        this.mSite = site;
+        this.mOnLoadMoreListener = onLoadMoreListener;
+        this.mCommentList = commentList;
+    }
+
+    @Override
+    public Fragment getItem(int position) {
+        return CommentDetailFragment.newInstance(mSite, getComment(position));
```

Comment:
I wonder if having each `CommentDetailFragment` store the  comment and site models in `onSaveInstanceState()` will result in that "transaction too large" issue described [here](https://issuetracker.google.com/issues/37103380#comment77). What do you think about storing only the comment ID and site ID to reduce the memory consumed by each fragments saved state?

Reply:
This totally makes sense, I'll change it.

Reply:
Do you think we need to load the comment asynchronously? It'd be loading it from local cache so it should be pretty fast (and the not is also being loaded synchronously) but my instinct would be to do it async.